yt/data_objects/time_series.py

```diff
@@ -56,16 +57,14 @@ def _eval(params, ds):
                 dict(eval = _eval, _params = tuple()))
     return cls
 
-def get_filenames_from_glob_pattern(filenames):
-    file_list = glob.glob(filenames)
-    if len(file_list) == 0:
-        data_dir = ytcfg.get("yt", "test_data_dir")
-        pattern = os.path.join(data_dir, filenames)
-        td_filenames = glob.glob(pattern)
-        if len(td_filenames) > 0:
-            file_list = td_filenames
-        else:
-            raise YTOutputNotIdentified(filenames, {})
+def get_filenames_from_glob_pattern(pattern):
+    epattern = os.path.expanduser(pattern)
+    data_dir = ytcfg.get("yt", "test_data_dir")
+    # if not match if found from the current work dir,
+    # we try to match the pattern from the test data dir
+    file_list = glob.glob(epattern) or glob.glob(os.path.join(data_dir, pattern))
+    if not file_list:
+        raise FileNotFoundError("No file matched this pattern or pattern(s) {}".format(pattern))
```

Comment:
Note on compatibility with #2695 (assuming both PRs are accepted)
`FileNotFoundError` subclasses `IOError` so : 
```python

try:
    yt.load("not_a_file*")
except IOError:
    print("couldn't load")
```
will work as intended and be consistent with non-pattern-like path.